presto-hive/src/test/java/com/facebook/presto/hive/TestHivePushdownFilterQueries.java

```diff
@@ -132,6 +132,30 @@ public void testNumeric()
         assertQueryUsingH2Cte("SELECT linenumber, ship_by_air, is_returned FROM lineitem_ex WHERE orderkey < 30000 AND ship_by_air = true");
     }
 
+    @Test
+    public void testDouble()
+    {
+        assertQuery("SELECT quantity, extendedprice, discount, tax from lineitem");
```

Comment:
you should use assertQueryUsingH2Cte as we are comparing the hive read with the h2 read.

Reply:
Like this? 
```
        // Nulls
        assertQueryUsingH2Cte("SELECT count(*) FROM lineitem_ex WHERE ship_day is null");

        assertQueryUsingH2Cte("SELECT sum(quantity) FROM lineitem_ex WHERE ship_by_air is null");

        assertQueryUsingH2Cte("SELECT sum(quantity) FROM lineitem_ex WHERE ship_by_air is null");

        assertQueryUsingH2Cte("SELECT quantity * (1 - discount) * (1 + tax) FROM lineitem_ex WHERE is_returned = true");
```

Reply:
You need to also generate null data for the queries to work. similar to how it was generated for Boolean and Byte readers
```
CASE WHEN linenumber % 5 = 0 THEN null ELSE shipmode = 'AIR' END AS ship_by_air
```

Reply:
Resolved, right?